l10n_it_intrastat_statement/models/intrastat_statement_sale_section.py

```diff
@@ -113,10 +113,13 @@ def _prepare_statement_line(self, inv_intra_line, statement_id=None):
             statement_id.company_id or company_id,
             dp_model.precision_get('Account'))
 
+        # check if additional_units has a value
+        has_additional_units = bool(inv_intra_line.additional_units)
         res.update({
             'transaction_nature_id': transaction_nature_id.id,
             'weight_kg': round(inv_intra_line.weight_kg) or 1,
-            'additional_units': round(inv_intra_line.additional_units) or 1,
+            'additional_units': round(inv_intra_line.additional_units) or (
+                0 if not has_additional_units else 1),
```

Comment:
questa non è equivalente a: `round(inv_intra_line.additional_units) or 0` ?
La logica è inserire 0 se non c'è l'unità aggiuntiva e l'unità arrotondata in caso contrario mi par di capire, in che caso ci va 1?

Reply:
in verità no, additional_units è un float e con valori tra 0.1 e 0.4 il round ritorna 0 (come in descrizione)

Reply:
**round(inv_intra_line.additional_units) or 1** porta a 1 anche se il valore è 0.0 e questo è sbalgiato

Reply:
Sergio l'arrotondamento dovrebbe esserci solo "se ricorre il caso", cioè quando la nomenclatura richieda una udm supplementare.
Se il ricalcolo mi fa valorizzare un campo che andrebbe valorizzato a 0 potrei avere una dichiarazione sbagliata ogni volta che lancio un ricalcolo.

Reply:
Quindi il round risulta vero solo se restituisce 1, in caso contrario risulta 0 se non esiste u.m. agg. e 1 in caso contrario (che vuol dire è inferiore a 0.5), mi pare ok.
Grazie